chebi-update/src/main/java/org/reactome/release/chebiupdate/ChebiUpdater.java

```diff
@@ -115,76 +152,112 @@ public void updateChebiReferenceMolecules() throws SQLException, Exception
+			reportIfMoleculeIdentifierChanged(molecule, chebiID, moleculeIdentifier, prefix);
+			boolean nameUpdated = updateMoleculeName(molecule, chebiName, moleculeName, prefix);
+			boolean formulaUpdated = updateMoleculeFormula(molecule, chebiFormulae, moleculeFormulae, prefix);
+			// If the ChEBI name was updated, update the referenceEntities that refer to the molecule.
+			if (nameUpdated)
+			{
+				updateReferenceEntities(molecule, chebiName, instanceEdit);
+			}
+			// Now, check to see if we need to update the ReferenceMolecule itself.
+			if (nameUpdated || formulaUpdated)
 			{
 				addInstanceEditToExistingModifieds(instanceEdit, molecule);
-				
-				if (adaptor.supportsTransactions())
-				{
-					adaptor.commit();
-				}
+				// Update the display name.
+				InstanceDisplayNameGenerator.setDisplayName(molecule);
+				adaptor.updateInstanceAttribute(molecule, ReactomeJavaConstants._displayName);
 			}
 		}
+		if (!testMode)
+		{
+			adaptor.commit();
+		}
+		else
+		{
+			adaptor.rollback();
+		}
+		
 		logger.info("*** Formula-fill changes ***");
 		logger.info(this.formulaFillSB.toString());
 		logger.info("*** Formula update changes ***");
 		logger.info(this.formulaUpdateSB.toString());
-		logger.info("*** Name update changes ***");
-		logger.info(this.nameSB.toString());
-		logger.info("*** Identifier update changes ***");
-		logger.info(this.identifierSB.toString());
+		
+		reportReferenceEntityChanges();
+	}
+
+	private void reportReferenceEntityChanges()
+	{
+		refEntChangeLog.info("# DB_ID\tCreator\tAffected ReferenceEntity\tNew ChEBI Name\tUpdated list of all names");
+		// Print the referenceEntities that have changes, sorted by who created them.
+		for (GKInstance creator : this.referenceEntityChanges.keySet().stream().sorted(this.personComparator).collect(Collectors.toList()))
+		{
+			for (String message : this.referenceEntityChanges.get(creator))
+			{
+				refEntChangeLog.info("{}\t{}",creator.toString(), message);
+			}
+		}
 	}
 
 	/**
-	 * Update a molecule's formula.
+	 * Update a molecule's formula. Should always update (set to the ChEBI formula) if the current formula and the formula from ChEBI differ.
 	 * 
 	 * @param molecule
 	 * @param chebiFormulae
 	 * @param moleculeFormulae
 	 * @param prefix
+	 * @return true if the formula was updated, or if the field was populated (i.e. was NULL before). False, otherwise.
 	 * @throws InvalidAttributeException
 	 * @throws InvalidAttributeValueException
 	 * @throws Exception
 	 */
-	private void updateMoleculeFormula(GKInstance molecule, List<DataItem> chebiFormulae, String moleculeFormulae, String prefix) throws InvalidAttributeException, InvalidAttributeValueException, Exception
+	private boolean updateMoleculeFormula(GKInstance molecule, List<DataItem> chebiFormulae, String moleculeFormulae, String prefix) throws InvalidAttributeException, InvalidAttributeValueException, Exception
 	{
+		boolean updated = false;
 		if (!chebiFormulae.isEmpty())
 		{
```

Comment:
You could invert this to use a guard clause so the method logic doesn't need to be in an if statement:

if (chebiFormula.isEmpty()) {
    return false;
}
...rest of the method

Reply:
I find that programming style confusing as the exit point is not obvious. I admit I do that sometimes but I try to avoid it when possible.